server/semantics-service/src/main/java/edu/illinois/ncsa/incore/service/semantics/controllers/TypeController.java

```diff
@@ -56,35 +55,53 @@ public class TypeController {
     @Inject
     private ISpaceRepository spaceRepository;
 
+    @Inject
+    private IAuthorizer authorizer;
+
     @Inject
     public TypeController(
         @Parameter(name = "User credentials.", required = true) @HeaderParam("x-auth-userinfo") String userInfo,
         @Parameter(name = "User groups.", required = false) @HeaderParam("x-auth-usergroup") String userGroups
     ) {
         this.username = UserInfoUtils.getUsername(userInfo);
         this.groups = UserGroupUtils.getUserGroups(userGroups);
-        // we want to limit the semantics service to admins for now
-        this.authorizer = new Authorizer();
-        if (!this.authorizer.isUserAdmin(this.groups)) {
-            throw new IncoreHTTPException(Response.Status.FORBIDDEN, this.username + " is not an admin.");
-        }
     }
 
     @GET
     @Path("types")
     @Produces({MediaType.APPLICATION_JSON})
     @Operation(summary = "list all types belong user has access to.")
-    public Response listTypes() {
+    public Response listTypes(
+        @Parameter(name = "Specify the order of sorting, either ascending or descending.") @DefaultValue("asc") @QueryParam("order") String order,
+        @Parameter(name = "Skip the first n results.") @DefaultValue("0") @QueryParam("skip") int offset,
+        @Parameter(name = "Limit number of results to return.") @DefaultValue("50") @QueryParam("limit") int limit,
+        @Parameter(name = "List the hyperlinks.") @DefaultValue("false") @QueryParam("hyperlink") boolean hyperlink) {
+        Comparator<String> comparator = Comparator.naturalOrder();
+        if (order.equals("desc")) comparator = comparator.reversed();
+
         List<Document> typeList = this.typeDAO.getTypes();
-        Set<String> userMembersSet = authorizer.getAllMembersUserHasReadAccessTo(username, spaceRepository.getAllSpaces(), groups);
-        //return the intersection between all types and the ones the user can read
-        List<Document> results = typeList.stream()
-            .filter(type -> userMembersSet.contains(type.getObjectId("_id").toString()))
+        List<String> results = typeList.stream()
+            .map(t -> t.get("dc:title").toString())
+            .sorted(comparator)
+            .skip(offset)
+            .limit(limit)
             .collect(Collectors.toList());
 
+        if (hyperlink) {
+            String typeEndpoint = "http://localhost:8080/";
```

Comment:
I would assume everything will be deployed under the same domain so we might be able to just use relative URL? 
no need to set the base URL then?
@navarroc @Rashmil-1999 is my understanding correct?

Reply:
For local testing, we will need to set `SERVICES_URL` in the local environment otherwise it gets Null only. I think that's why other services set base URL (String typeEndpoint = "http://localhost:8080/")

Reply:
Yes, in terms of code what I meant to have was directly use the hostname env variable like how we have in maestro and frontend applications so that the code can be dynamic, rather than static....

Reply:
I just looked at it a bit...
We can probably remove string endpoint="http://localhost:8080/" from codes as long as the env is set. 
Most of time we use IDE for runs, and IDE has all the env variables set up already. Maybe we can chat about this in meetings.

But if we make this change, should all the other services do the same?

Reply:
Other services might need that as a fallback? we can discuss that more in the next meeting

Reply:
As other services have the same/similar patterns, I think it can have another issue if we decide to do it.

Reply:
agree!